fennel/test_lib/mock_client.py

```diff
@@ -740,6 +744,43 @@ def lookup(
         )
         return data[fields].to_dict(orient="records"), found
 
+    def erase_key(self, dataset_name: str, erase_keys: List[Dict[str, Any]]):
+
+        try:
+            dataset_info = self.dataset_info[dataset_name]
+            data = self.data[dataset_name]
+        except KeyError:
+            raise KeyError(f"Dataset: {dataset_name} not found")
+
+        for erase_key in erase_keys:
+            for field_name in erase_key.keys():
+                if field_name not in dataset_info.fields:
+                    raise ValueError(f"Field: {field_name} not in dataset")
+                if field_name not in dataset_info.key_fields:
+                    raise ValueError(
+                        f"Field: {field_name} not in a key field in dataset"
+                    )
+                if field_name not in dataset_info.erase_key_fields:
+                    raise ValueError(
+                        f"Field: {field_name} is not an erase key field in dataset"
+                    )
```

Comment:
We should also test that ALL erase keys have been provided, right? Is that being done somewhere?

Reply:
Good catch